### PR TITLE
feat:[CI-10093]:Add-http-porxy-support-drone-git

### PR DESCRIPTION
**Pull request by HenAmar**

### CODE REVIEW

File: posix/clone-commit [hunk 1]
```diff
@@ -12,6 +12,12 @@ if [ ! -d .git ]; then
 	set +x
 fi
 
+if [ -n "$DRONE_HTTP_PROXY_URL" ] && [ -n "$DRONE_HTTP_PROXY_PORT" ]; then
```
Comment (devkimittal) [inline posix/clone-commit:15]: is this working, i think it should be written like  "${DRONE_HTTP_PROXY_URL}"

File: windows/clone-tag.ps1 [hunk 2]
```diff
@@ -20,6 +20,11 @@ if (!(Test-Path .git)) {
 	iu git remote add origin "$Env:DRONE_REMOTE_URL"
 }
 
+if (![string]::IsNullOrEmpty($env:DRONE_HTTP_PROXY_URL) -and ![string]::IsNullOrEmpty($env:DRONE_HTTP_PROXY_PORT)) {
+    Write-Host "+ git config --global --global http.proxy $Env:DRONE_HTTP_PROXY_URL:$Env:DRONE_HTTP_PROXY_PORT "
```
Comment (devkimittal) [inline windows/clone-tag.ps1:24]: let's put capital Env, everywhere? I have seen issues in env vs Env in windows

File: posix/clone-commit (hunk 1 above)
Comment (HenAmar) [reply posix/clone-commit:15]: @devkimittal yes it work 
image

Comment (devkimittal) [reply posix/clone-commit:15]: i see, have you tried an e2e run though? Also what about ssh clone, will it require additional changes as well?

Comment (HenAmar) [reply posix/clone-commit:15]: no, we don't support ssh via proxy for the moment. This is only http proxy. "git config --global http.proxy"

**Review by devkimittal — APPROVED** [review]

**Pull request merged**